File: backend/app/schemas/character.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CharacterRead(BaseModel):
    id: int
    slug: str
    name: str
    aliases: str | None = None
    description: str | None = None
    avatar_url: str | None = None
    theme_color: str | None = None
    sort_order: int
    is_featured: bool
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class CharacterTimelineEvent(BaseModel):
    date: str
    title: str
    content: str


class CharacterDetailRead(CharacterRead):
    avatar_large_url: str | None = None
    origin_story: str | None = None
    timeline: list[CharacterTimelineEvent] | None = None

File: frontend/src/types/character.ts
export interface Character {
  id: number
  slug: string
  name: string
  aliases: string | null
  description: string | null
  avatar_url: string | null
  theme_color: string | null
  sort_order: number
  is_featured: boolean
  created_at: string
  updated_at: string
}

export interface CharacterTimelineEvent {
  date: string
  title: string
  content: string
}

export interface CharacterDetail extends Character {
  avatar_large_url: string | null
  origin_story: string | null
  timeline: CharacterTimelineEvent[] | null
}
